src/index.ts

```diff
@@ -569,3 +585,40 @@ const main = async ({
     })
   );
 })();
+
+type PackageManager = "npm" | "pnpm" | "yarn";
+
+async function exists(path) {
+  try {
+    await access(path);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function getPkgManager(): Promise<PackageManager> {
+  // Use npm_config_user_agent if it's set
+  const userAgent = process.env.npm_config_user_agent;
+
+  if (userAgent) {
+    if (userAgent.startsWith("yarn")) {
+      return "yarn";
+    } else if (userAgent.startsWith("pnpm")) {
+      return "pnpm";
+    } else {
+      return "npm";
+    }
+  }
+
+  const hasYarnLock = await exists("yarn.lock");
```

Comment:
I'd do `if (hasYarnLock) return "yarn"` just after this line to avoid unnecessary checking for pnpm-lock.